torch/nn/functional.py

```diff
@@ -565,3 +565,32 @@ def pad(input, pad, mode='constant', value=0):
             return _functions.thnn.ReplicationPad3d(*pad)(input)
     else:
         raise NotImplementedError("Only 4D and 5D padding is supported for now")
+
+
+# distance
+
+def pairwise_distance(x1, x2, p=2, eps=1e-6):
+    r"""
+    Computes the batchwise pairwise distance between vectors v1,v2:
+
+        .. math ::
+            \Vert x \Vert _p := \left( \sum_{i=1}^n  \vert x_i \vert ^ p \right) ^ {1/p}
+
+        Args:
+            x (Tensor): input tensor containing the two input batches
+            p (real): the norm degree. Default: 2
+
+        Shape:
+            - Input: :math:`(N, D)` where `D = vector dimension`
+            - Output: :math:`(N, 1)
+
+        >>> input1 = autograd.Variable(torch.randn(100, 128))
+        >>> input2 = autograd.Variable(torch.randn(100, 128))
+        >>> output = F.pairwise_distance(input1, input2, p=2)
+        >>> output.backward()
+    """
+    assert x1.size() == x2.size(), "Input sizes must be equal."
+    assert x1.dim() == 2, "Input rank must be equal to two."
```

Comment:
Well, number of dimensions isn't really a matrix rank 😕 I'd just say that it should be a 2D tensor.